src/main/java/org/broadinstitute/hellbender/utils/fermi/Assembly.java

```diff
@@ -0,0 +1,92 @@
+package org.broadinstitute.hellbender.utils.fermi;
+
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+
+import java.io.*;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+/** an assembly is just a collection of contigs */
+public final class Assembly {
+    private final List<Contig> contigs;
+
+    Assembly( final List<Contig> contigs ) { this.contigs = contigs; }
+
+    public int getNContigs() { return contigs.size(); }
+    public Contig getContig( final int idx ) { return contigs.get(idx); }
+    public List<Contig> getContigs() { return Collections.unmodifiableList(contigs); }
+
+    /** a sequence of bases, coverage data, and connections to other contigs */
+    public final static class Contig {
+        private final byte[] sequence;
+        private final int nSupportingReads;
+        private final byte[] perBaseCoverage;
+        private List<Connection> connections;
+
+        Contig( final byte[] sequence, final int nSupportingReads, final byte[] perBaseCoverage ) {
+            this.sequence = sequence;
+            this.nSupportingReads = nSupportingReads;
+            this.perBaseCoverage = perBaseCoverage;
+        }
+
+        public byte[] getSequence() { return sequence; }
+        public int getNSupportingReads() { return nSupportingReads; }
+        public byte[] getPerBaseCoverage() { return perBaseCoverage; }
+        public List<Connection> getConnections() { return Collections.unmodifiableList(connections); }
+
+        void setConnections( final List<Connection> connections ) {
+            this.connections = connections;
+        }
+    }
+
+    /** a connection between contigs */
+    public final static class Connection {
+        private final Contig target;      // contig that overlaps the one that possesses this connection
+        private final int overlapLen;     // bases in common -- negative overlap lengths are legal, and represent gaps
```

Comment:
Yay, gaps are allowed! A related question: how difficult do you think it would be to have a CIGAR embedded in this Connection? I know fermilite doesn't seem to do this right now, but I'm just curious.